pkg/k8s/kubernetes_exec.go

```diff
@@ -13,13 +14,38 @@ import (
 	execute "github.com/alexellis/go-execute/pkg/v1"
 )
 
+// Capabilities is an index of the support API versions on the server
+type Capabilities map[string]bool
+
 func GetNodeArchitecture() string {
 	res, _ := KubectlTask("get", "nodes", `--output`, `jsonpath={range $.items[0]}{.status.nodeInfo.architecture}`)
 
 	arch := strings.TrimSpace(string(res.Stdout))
 
 	return arch
 }
+
+// GetCapabilities returns the supported API versions on the server
+func GetCapabilities() (Capabilities, error) {
```

Comment:
Interesting approach. I quite like that 💪

Reply:
I was inspired by a similar feature in helm and I thought this was cleaner than checking semver. It is directly answering the question "does this cluster support this api" instead of depending on us reading documentation and changelogs correctly to write the correct semver requirement and then documenting that decision in our own code.